systems/plants/RigidBody.h

```diff
@@ -44,23 +44,31 @@ class DLLEXPORT_RBM RigidBody {
   int robotnum; // uses 0-index. starts from 0
   static const std::set<int> defaultRobotNumSet;
 // note: it's very ugly, but parent,dofnum,and pitch also exist currently (independently) at the rigidbodymanipulator level to represent the featherstone structure.  this version is for the kinematics.
-  int parent;
+  std::shared_ptr<RigidBody> parent;
+  int body_index; // index in RBM bodies vector (set in compile()) TODO: remove after we finish the conversion from parents being body indices to being pointers
   int dofnum; // interpreted as start of position_num from Matlab
   int velocity_num_start;
   int floating; // FLOATINGBASE TODO: remove
   int pitch; // FLOATINGBASE TODO: remove
+  Matrix4d Ttree;  // floatingbase TODO: replace with Isometry3d?
+  Matrix4d T_body_to_joint;  // floatingbase TODO: replace with Isometry3d?
+
+  typedef struct _CollisionElement {
+    DrakeCollision::Shape shape;
+    std::vector<double> params;
+    Isometry3d T;
+  } CollisionElement;
+  std::vector<CollisionElement> collision_elements;
```

Comment:
collision elements now have to get added at compile time, since the body indices may change.  so i store the collision info from the parser here.  (just like in matlab)